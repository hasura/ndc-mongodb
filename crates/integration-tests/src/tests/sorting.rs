use insta::assert_yaml_snapshot;

use crate::graphql_query;

#[tokio::test]
async fn sorts_on_extended_json() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
                query Sorting {
                  extendedJsonTestData(order_by: { value: Desc }) {
                    type
                    value
                  }
                }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}

#[tokio::test]
async fn sorts_on_nested_field_names_that_require_escaping() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
            query {
              testCases_weirdFieldNames(limit: 1, order_by: { invalidName: Asc }) {
                invalidName
                invalidObjectName {
                  validName
                }
                validObjectName {
                  invalidNestedName
                }
              }
            }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}
