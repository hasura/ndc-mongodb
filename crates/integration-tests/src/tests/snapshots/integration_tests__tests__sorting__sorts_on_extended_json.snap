---
source: crates/integration-tests/src/tests/sorting.rs
expression: "graphql_query(r#\"\n                query Sorting {\n                  extendedJsonTestData(order_by: { value: Desc }) {\n                    type\n                    value\n                  }\n                }\n            \"#).run().await?"
---
data:
  extendedJsonTestData:
    - type: date
      value:
        $date:
          $numberLong: "1724164680000"
    - type: date
      value:
        $date:
          $numberLong: "1637571600000"
    - type: string
      value: "hello, world!"
    - type: string
      value: foo
    - type: long
      value:
        $numberLong: "8"
    - type: long
      value:
        $numberLong: "7"
    - type: int
      value:
        $numberInt: "6"
    - type: int
      value:
        $numberInt: "5"
    - type: double
      value:
        $numberDouble: "4.0"
    - type: double
      value:
        $numberDouble: "3.0"
    - type: decimal
      value:
        $numberDecimal: "2"
    - type: decimal
      value:
        $numberDecimal: "1"
    - type: "null"
      value: ~
errors: ~
